Content.Server/_RMC14/Xenonids/Weeds/XenoWeedsSystem.cs

```diff
@@ -61,7 +61,8 @@ private void OnWeedsSpreadNeighbors(Entity<XenoWeedsComponent> ent, ref SpreadNe
             if (diff.X >= ent.Comp.Range || diff.Y >= ent.Comp.Range)
                 break;
 
-            if (!CanPlaceWeeds((gridOwner, neighbor.Grid), tile))
+            if (!CanPlaceWeeds((gridOwner, neighbor.Grid), tile, out var semiWeedable, false) &&
+                !(semiWeedable && ent.Comp.SpreadsOnSemiWeedable))
```

Comment:
This semi weeedable check should be done by canplaceweeds, you can add a boolean to it for spreads on semi weedable then pass that in

Reply:
Moved the check to CanPlaceWeeds